app/src/main/java/io/appium/uiautomator2/handler/FindElements.java

```diff
@@ -170,7 +170,7 @@ protected AppiumResponse safeHandle(IHttpRequest request) throws UiObjectNotFoun
         boolean keepSearching = true;
```

Comment:
It would be nice to extract this helper into a separate class and cover it with unit tests. Ofc not in this PR

Do you have time to help with that @robertstreche ?

Reply:
The PR [https://github.com/appium/appium-uiautomator2-server/pull/411](url) seems to be just the refactoring you are talking about.

Reply:
yes, I wanted to do such refactoring long time ago, just did not have a good test suite to verify it on.